tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala

```diff
@@ -174,9 +180,9 @@ class IndentWhenPastingSuite
        |object Main:
        |  object SubMain:
        |    enum                        Color(val rgb: Int):
-       |      case Red   extends Color(0xFF0000)
-       |      case Green extends Color(0x00FF00)
-       |      case Blue  extends Color(0x0000FF)
+       |       case Red   extends Color(0xFF0000)
```

Comment:
In pasted code it was 3.
Previously it was corrected because of implementations specifics but it was rather a corner case than general logic.

Reply:
Pasted code was 3, but we adjsut it to fit how the current document is formatted.

Reply:
We shouldn't touch it because it's relative indentation in pasted block .
The only we could do in this formatter is to increase/decrease indentation for the whole pasted block.

Examples:
```scala
// pasted code
Color:
     case Red extends Color(0xFF0000)

// paste - increase indentation by 2 on each line
object Main: 
  Color:                                                   
      case Red extends Color(0xFF0000)
     
// or if pasted block was
      Color:
           case Red extends Color(0xFF0000)

// paste - decrease indetation by 4 on each line
object Main: 
  Color:                                                    
      case Red extends Color(0xFF0000)
      
// we should keep the original indentation between pasted lines
// otherwise we might drop significant indentation or uglify code
val a =
  2

// paste -  how it shouldn't work
object Main:
  val a = 
  2          
```

Reply:
I am not talking here about dropping the indentation, but rather if we detect the diff in indentation between consecutive lines in the pasted fragment to be different than the current indentation, we could change it to the current indentation.

Reply:
I think that was the intention before, that we would adjust the indentation.

Reply:
Previously it worked only for the second line in this case because of implementation specifics. 

For example if you paste this sample into current implementation - the second line indent will be reduced  from 3 to 2 but the third one keep the relative difference 3. 
<pre><code>
Color:
&nbsp;&nbsp;&nbsp;case Red
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;case Black
</code></pre>

I would prefer not to try to correct relative indentation here.
It seems that it was a reason why multiline string pasting wasn't perfect in this case (now it's fixed): https://github.com/scalameta/metals/blob/2adcb3c47e923cd78492e3845d701ef3b9600fb1/tests/unit/src/test/scala/tests/rangeFormatting/MultilineStringRangeFormattingWhenPastingSuite.scala#L330-L332